app/lib/grades/grades_service/grades_service.dart

```diff
@@ -680,7 +689,15 @@ class Grade {
   });
 }
 
-enum WeightType { perGrade, perGradeType, inheritFromTerm }
+enum WeightType {
+  perGrade,
+  perGradeType,
+  inheritFromTerm;
+
+  static WeightType fromString(String value) {
+    return values.firstWhere((e) => e.name == value);
+  }
```

Comment:
Why not `WeightType.values.byName(value)` in the code where you call `WeightType.fromString(value)`?

Reply:
I forgot that this method exists, thanks for the reminder 👍